### PR TITLE
refactor code to use bpftrace as function call

### PR DESCRIPTION
**Pull request by yunwei37**
- [x] use function call styles for bpftrace
-  [x] refactor the prompts and chain
- [x] remove generate ebpf program option
 -  [x] remove train related api because we will use embedding examples
 - [x] add new embbeding example

### CODE REVIEW

File: gpttrace/bpftrace.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            # Make sure the timer is canceled
+            timer.cancel()
+            return {
+                "command": ' '.join(command),
+                "stdout": stdout,
+                "stderr": stderr,
+                "returncode": process.returncode
+            }
+
+def construct_command(operation):
```
Comment (Officeyutong) [inline gpttrace/bpftrace.py:116]: Typing hints for this function?

**Review by Officeyutong — COMMENTED** [review]
May be all functions should be added with typing hints for better maintainability, since this is a long-term project instead of a poc

File: gpttrace/bpftrace.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                    "description": "the file content"
+                }
+            },
+            "required": ["filename", "continue"]
+        }
+    }
+]
+
+
+def run_command_with_timeout(command: str, timeout: int) -> dict:
```
Comment (Officeyutong) [inline gpttrace/bpftrace.py:99]: Consider TypedDict if the return value is a dict with fixed keys and value types

**Review by Officeyutong — COMMENTED** [review]
typing hints should be more concrete (i.e concrete types or TypedDicts, instead of general dict/list/object). You can use mypy to run static type check on your code

File: gpttrace/bpftrace.py (hunk 2 above)
Comment (yunwei37) [reply gpttrace/bpftrace.py:99]: Thanks! it's added.

**Pull request merged**